Source/WebCore/layout/formattingContexts/inline/InlineItemsBuilder.cpp

```diff
@@ -745,6 +745,21 @@ static inline bool canCacheMeasuredWidthOnInlineTextItem(const InlineTextBox& in
 void InlineItemsBuilder::computeInlineTextItemWidths(InlineItemList& inlineItemList)
 {
     TextSpacing::SpacingState spacingState;
+    TrimmableTextSpacings trimmableTextSpacings;
+    auto handleTextSpacing = [&](const InlineTextItem& inlineTextItem, size_t inlineItemIndex) {
+        const auto& autospace = inlineTextItem.style().textAutospace();
+        auto& content = inlineTextItem.inlineTextBox().content();
+        if (!autospace.isNoAutospace()) {
+            // We need to store information about spacing added between inline text items since it needs to be trimmed during line breaking if the consecutive items are placed on different lines
+            auto characterClass = TextSpacing::characterClass(content.characterAt(inlineTextItem.start()));
+            if (autospace.shouldApplySpacing(spacingState.lastCharacterClassFromPreviousRun, characterClass))
+                trimmableTextSpacings.add(inlineItemIndex, autospace.textAutospaceSize(inlineTextItem.style().fontCascade().primaryFont()));
+
+            spacingState.lastCharacterClassFromPreviousRun = TextSpacing::characterClass(content.characterAt(inlineTextItem.start() + inlineTextItem.length() - 1));
+        } else
+            spacingState.lastCharacterClassFromPreviousRun = TextSpacing::CharacterClass::Undefined;
+    };
+
```

Comment:
This functions is getting dominated by textspacing. Maybe it's time to factor it out to its own function.